pkg/river/encoding/attribute.go

```diff
@@ -0,0 +1,61 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/agent/pkg/river/internal/rivertags"
+	"github.com/grafana/agent/pkg/river/internal/value"
+)
+
+// attributeField represents a json representation of a river attribute.
+type attributeField struct {
+	field    `json:",omitempty"`
+	valField riverField
+}
+
+func newAttribute(val value.Value, f rivertags.Field) (*attributeField, error) {
+	af := &attributeField{}
+	return af, af.convertAttribute(val, f)
+}
+
+func (af *attributeField) isValid() bool {
+	if af == nil || af.valField == nil {
+		return false
+	}
+	return af.valField.hasValue()
+}
```

Comment:
Did you mean to rename this to hasValue rather than isValid? There's a few other methods still named isValid too.